Frameworks/include/CGIWICBitmap.h

```diff
@@ -60,8 +62,15 @@ class CGIWICBitmapLock : public RuntimeClass<RuntimeClassFlags<RuntimeClassType:
         m_bytesPerRow = static_cast<size_t>(bpr);
     }
 
+    CGIWICBitmapLock(_In_ BYTE* data, _In_ const WICRect* region, _In_ size_t bytesPerRow, _In_ WICPixelFormatGUID pixelFormat)
+        : m_dataBuffer(data), m_pixelFormat(pixelFormat), m_locked_rect(region), m_bytesPerRow(bytesPerRow) {
+        m_texture = nullptr;
+    }
+
     ~CGIWICBitmapLock() {
-        m_texture->Unlock();
+        if (m_texture) {
+            m_texture->Unlock();
+        }
         m_texture = nullptr;
```

Comment:
whats going on here? You have a raw* but aren't jacketing in an RAII appropriate construct. Ownership is confusing and certainly not thread safe as is.